### PR TITLE
Allow passing a Redis instance to Percy::Hub.new

### PR DESCRIPTION
**Pull request by faun**
Changes:

- Rename RedisService to RedisConnection and convert from a module to a class.

- Instantiate a RedisConnection object during percy-hub initialization instead of including RedisService module.

- Allow passing an existing instance of Redis to percy-hub initialization, enabling connection re-use at a higher-level than each instance. This will enable us to pass in a [ConnectionPool::Wrapper](https://github.com/mperham/connection_pool/blob/master/lib/connection_pool/wrapper.rb) instance that will allow the client to opt-in to connection pooling.

### CODE REVIEW

File: lib/percy/hub/redis_connection.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          @redis_options = configure_redis_options(options)
+        end
+      end
+
+      def redis
+        @redis ||= redis_connection.client
+      end
+
+      def disconnect_redis
+        return if connection_pool?
```
Comment (fotinakis) [inline lib/percy/hub/redis_connection.rb:29]: Need an `attr_reader` for this?

**Review by fotinakis — APPROVED** [review]
LGTM 🌮  one question/comment

File: lib/percy/hub/redis_connection.rb (hunk 1 above)
Comment (faun) [reply lib/percy/hub/redis_connection.rb:29]: It's an actual method: https://github.com/percy/percy-hub/pull/47/files/ad7f94aff99ff8c26b85e76f3667598a3ea5ba4b#diff-99c0773ea4107e7c191b8c3a71e1eb133c773f5eb06f8349c3ec8699ca4a8b9fR75

Comment (fotinakis) [reply lib/percy/hub/redis_connection.rb:29]: 👍

**Pull request merged**